File: quick-sort/quick_sort.php
<?php
function quickSort($arr) {
    if (count($arr) <= 1) return $arr;
    
    $pivot = $arr[0];
    $less = $greater = [];
    
    for ($i = 1; $i < count($arr); $i++) {
        if ($arr[$i] < $pivot) {
            $less[] = $arr[$i];
        } else {
            $greater[] = $arr[$i];
        }
    }
    
    return array_merge(quickSort($less), [$pivot], quickSort($greater));
}

$arr = [64, 34, 25, 12, 22, 11, 90];
print_r(quickSort($arr));
?>

File: quick-sort/quick_sort.clj
(defn quick-sort [coll]
  (if (<= (count coll) 1)
    coll
    (let [pivot (first coll)
          rest-coll (rest coll)
          less (filter #(< % pivot) rest-coll)
          greater (filter #(>= % pivot) rest-coll)]
      (concat (quick-sort less) [pivot] (quick-sort greater)))))

(println (quick-sort [64 34 25 12 22 11 90]))